scheduler/generic_sched.go

```diff
@@ -125,7 +125,14 @@ func NewBatchScheduler(logger log.Logger, eventsCh chan<- interface{}, state Sta
 }
 
 // Process is used to handle a single evaluation
-func (s *GenericScheduler) Process(eval *structs.Evaluation) error {
+func (s *GenericScheduler) Process(eval *structs.Evaluation) (err error) {
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("processing eval %q panicked scheduler: %v", eval.ID, r)
```

Comment:
Either here or where it gets logged can we throw a `This is a bug!` in the message to let operators know this isn't something they have control over and should be reported.

Reply:
We must be sharing some brain-waves b/c I was totally thinking that but thought I was being too "extra". 😀 Will do!

Reply:
Done in [f13c7be](https://github.com/hashicorp/nomad/pull/12009/commits/f13c7beea62df33be2fb54fb84ac438b85b5e72e)